controller/api/destination/server.go

```diff
@@ -164,16 +164,50 @@ func (s *server) GetProfile(dest *pb.GetDestination, stream pb.Destination_GetPr
 		return status.Errorf(codes.InvalidArgument, "invalid authority: %s", err)
 	}
 
+	var service watcher.ServiceID
+	var path string
+
 	if ip := net.ParseIP(host); ip != nil {
-		// TODO handle lookup by IP address
-		log.Debug("Lookup of IP addresses is not supported for profiles")
-		return status.Errorf(codes.InvalidArgument, "cannot discover profiles for IP addresses")
-	}
+		svc, exists, err := s.ips.GetSvc(ip.String())
+		if err != nil {
+			return err
+		}
+		if exists {
+			service = *svc
+			path = fmt.Sprintf("%s.%s.svc.%s", service.Name, service.Namespace, s.clusterDomain)
+		} else {
+			primary, secondary := newFallbackProfileListener(translator)
+
+			if dest.GetContextToken() != "" {
+				err = s.profiles.Subscribe(nil, primary)
+				if err != nil {
+					log.Warnf("Failed to subscribe to profile %s: %s", path, err)
+					return err
+				}
+				defer s.profiles.Unsubscribe(nil, primary)
```

Comment:
I think the error that Unsubsribe can sometimes return will be muted when calling it via defer. Maybe better to leave the function without a return value and just log on error.

Reply:
Yes you are correct here that errors that occur in `Unsubscribe` will fail silently.

I tested this by inserting `return fmt.Errorf("this is an error")` in `Unsubscribe`, and then subscribed to and quit a profile stream.

I expected to see this error in the server, but nothing occurred.

This occurs in the other places that we unsubscribe in this file, so I lean towards making this change separately.